Index: aane-lifemap/sites/all/modules/advanced_help/advanced_help.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- aane-lifemap/sites/all/modules/advanced_help/advanced_help.module	(revision )
+++ aane-lifemap/sites/all/modules/advanced_help/advanced_help.module	(revision )
@@ -205,7 +205,7 @@
 
   $output = array();
   // Print a search widget.
-  $output['advanced_help_search'] = module_exists('search') ? drupal_get_form('advanced_help_search_form') : t('Enable the search module to search help.');
+  $output['advanced_help_search'] = module_exists('search') ? drupal_get_form('advanced_help_search_form') : array('#markup'=>t('Enable the search module to search help.'));
 
   $breadcrumb = array();
   if ($module) {
